### ISSUE TITLE
Number format settings don't work for remapped columns

### ISSUE THREAD

**Issue by noahmoss** [metabase/metabase#17427]
**Describe the bug**
If you save a custom mapping for a number column, the column settings sidebar still allows you to modify number format settings, but they have no effect on the values (even if the mapped values are unchanged, or are other numbers).

**To Reproduce**
1. Go to Admin -> Data Model -> Sample Dataset -> Orders 
2. Click the gear button next to "Quantity"
3. Under "Display Values" choose "Custom mapping", then click "Save" at the bottom of the list of values (without changing any of the mappings) 
4. Go to Ask a question -> Simple question -> Sample Dataset -> Orders -> Quantity -> click the gear icon
6. Modify any of these settings: "Style", "Separator style", "Minimum number of decimal places", and "Multiply by a number"

**Expected behavior**
Either we should hide the settings that are not supported for remapped columns, or we should support these settings for remapped values that are numbers.

**Screenshots**
<img width="462" alt="image" src="https://user-images.githubusercontent.com/32746338/129127748-e9db3fb8-8846-473a-a727-71b7c9a5921e.png">

**Information about your Metabase Installation:**

Current master

**Comment by flamber** [comment]
This is caused by the custom mapping always returning a string to be displayed, but the underlying value is still there, which is what is causing confusion like #11458.

**Comment by cdeweyx** [comment]
We should at the very least hide these settings, but it feels weird not to support this. @kulyk do you have any familiarity with how remapped data interacts with column formatting?

**Comment by lorem--ipsum** [comment]
I don't think it makes sense to allow numerical operations (such as multiplication, number of decimal places, etc.) on strings, which is what remapping a column implicitely does. What about something like this:

<img width="1590" alt="Image" src="https://github.com/user-attachments/assets/b15c0e49-0d80-438d-a8fa-03f74119bbcd" />

**Comment by perivamsi** [comment]
We should show the formatting settings based on the remapping column, not the original column
